File: src/main/java/uk/gov/justice/digital/hmpps/riskprofiler/datasourcemodel/FileType.java
package uk.gov.justice.digital.hmpps.riskprofiler.datasourcemodel;

import java.util.Arrays;

public enum FileType {
    PRAS(Pras.class),
    OCGM(OcgmList.class),
    OCG(Ocg.class),
    PATHFINDER(PathFinder.class),
    VIPER(Viper.class);

    private final Class<? extends RiskDataSet> type;

    FileType(Class<? extends RiskDataSet> type) {
        this.type = type;
    }

    public Class<? extends RiskDataSet> getType() {
        return type;
    }

    public static FileType byDataSet(Class<? extends RiskDataSet> clazz) {
        return Arrays.stream(FileType.values())
                .filter(ft -> ft.type == clazz)
                .findFirst().orElse(null);

    }

}

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/datasourcemodel/FileType.kt
package uk.gov.justice.digital.hmpps.riskprofiler.datasourcemodel

import java.util.Arrays

enum class FileType(val type: Class<out RiskDataSet?>) {
  PRAS(Pras::class.java), OCGM(OcgmList::class.java), OCG(Ocg::class.java), VIPER(Viper::class.java);

  companion object {
    @JvmStatic
    fun byDataSet(clazz: Class<out RiskDataSet?>): FileType {
      return Arrays.stream(values())
        .filter { ft: FileType -> ft.type == clazz }
        .findFirst().orElse(null)
    }
  }
}
